runtime/sema/access.go

```diff
@@ -458,6 +464,9 @@ func (a PrimitiveAccess) Equal(other Access) bool {
 }
 
 func (a PrimitiveAccess) PermitsAccess(otherAccess Access) bool {
+	if ast.PrimitiveAccess(a) == ast.AccessNone {
+		return PrimitiveAccess(ast.AccessNone) == otherAccess
```

Comment:
Avoid Yoda condition:

```suggestion
		return otherAccess == PrimitiveAccess(ast.AccessNone)
```